### ISSUE TITLE
Assess interpretability

### ISSUE THREAD

**Issue by gcastro-98** [gcastro-98/synthetic-medical-images#1]
Enhance the decoder CNN by explaining its decisions by:
- [ ] [**Layer Wise Relevance Propagation**](http://www.heatmapping.org/): there are several `pytorch` implementations. For instance:
  - [Tutorial](https://git.tu-berlin.de/gmontavon/lrp-tutorial) to apply them to VGG-16 and other arquitectures 
  - [A different flavour](https://github.com/fhvilshoj/TorchLRP) also appliable tto pre-trained large VGG models of `torchvision`
  - [Other different suggestions](https://discuss.pytorch.org/t/layer-wise-relevance-propagation-lrp-in-pytorch/46730): note residual connections (_e.g._ those of ResNet) may give continuity problems
- [ ] [Gradient-based localization (Grad-CAM)](https://arxiv.org/pdf/1610.02391.pdf):
  -  There are plenty of [guides](https://medium.com/@stepanulyanin/implementing-grad-cam-in-pytorch-ea0937c31e82) out there but [OmniXAI](https://opensource.salesforce.com/OmniXAI/latest/tutorials/vision/gradcam_torch.html) provides some interesting implementations
- Other explainability flavours **different than pixel-attribution** methods:
  - [ ] [Concept Activation Vectors (CAV)](https://paperswithcode.com/paper/interpretability-beyond-feature-attribution): check, for instance, [captum](https://github.com/pytorch/captum) implementation
  - [ ] [Counterfactual](https://opensource.salesforce.com/OmniXAI/latest/tutorials/vision/ce_torch.html) explanations
  - [ ] [Contrastive](https://opensource.salesforce.com/OmniXAI/latest/tutorials/vision/cem_torch.html) explanations
  - [ ] [LIME for image classification](https://opensource.salesforce.com/OmniXAI/latest/tutorials/vision/lime.html)
